lint: Remove epic and issue ref linter from cockroach repo

nickvigilante opened the pull request:
This PR is one of two PRs needed to move the epic and issue ref linter from the cockroach repo to Blathers.

This PR should not be merged until https://github.com/cockroachlabs/blathers-bot/pull/93 is merged.

Release note: None

Part of #90189

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94004)
<!-- Reviewable:end -->

Commit:
lint: Remove epic and issue ref linter from cockroach repo

This PR is one of two PRs needed to move the epic and issue ref linter
from the cockroach repo to Blathers.

This PR should not be merged until cockroachlabs/blathers-bot#92 is
merged.

Release note: None

Part of #90189

nickvigilante commented:
bors r+
